src/main/java/org/jetbrains/research/intellijdeodorant/util/math/Edge.java

```diff
@@ -0,0 +1,52 @@
+package org.jetbrains.research.intellijdeodorant.util.math;
+
+public class Edge implements Comparable<Edge> {
+
+    protected final Node from, to;
+    protected final int weight;
+    private volatile int hashCode = 0;
+
+    public Edge(final Node argFrom, final Node argTo, final int argWeight) {
+        from = argFrom;
+        to = argTo;
+        weight = argWeight;
+    }
+
+    public Node getSource() {
+        return from;
+    }
+
+    public Node getTarget() {
+        return to;
+    }
+
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+
+        if (o instanceof Edge) {
+            Edge edge = (Edge) o;
+            return this.from.equals(edge.from) && this.to.equals(edge.to);
+        }
+        return false;
+        // :thi
```

Comment:
?

Reply:
I honestly have no idea how and why this comment appeared here. Removed.